libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -673,6 +673,47 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
 bool IRGeneratorForStatements::visit(UnaryOperation const& _unaryOperation)
 {
 	setLocation(_unaryOperation);
+
+	if (FunctionDefinition const* function = _unaryOperation.annotation().userDefinedFunction)
+	{
+		_unaryOperation.subExpression().accept(*this);
+		setLocation(_unaryOperation);
+
+		solAssert(
+			dynamic_cast<SourceUnit const*>(function->scope()),
+			"Only file-level functions and library functions can be bound to a user type operator."
+		);
+
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+			function->libraryFunction() ? function->typeViaContractName() : function->type()
+		);
+		solAssert(functionType);
+		functionType = dynamic_cast<FunctionType const&>(*functionType).asBoundFunction();
+		solAssert(functionType);
+		solAssert(
+			functionType->parameterTypes().size() == 0,
+			"Functions with parameters other than self parameter cannot be bound to a user type unary operator."
+		);
+
+		string parameter = expressionAsType(_unaryOperation.subExpression(), *functionType->selfType());
+		solAssert(!parameter.empty());
```

Comment:
Are you sure it cannot be empty? At some point we might have types that do not occupy any stack slots.